src/main/java/net/dv8tion/jda/core/requests/Route.java

```diff
@@ -393,12 +369,36 @@ public String toString()
         private final Route baseRoute;
         private final String ratelimitRoute;
         private final String compiledRoute;
+        private final boolean hasQueryParams; 
 
-        private CompiledRoute(Route baseRoute, String ratelimitRoute, String compiledRoute)
+        private CompiledRoute(Route baseRoute, String ratelimitRoute, String compiledRoute, boolean hasQueryParams)
         {
             this.baseRoute = baseRoute;
             this.ratelimitRoute = ratelimitRoute;
             this.compiledRoute = compiledRoute;
+            this.hasQueryParams = hasQueryParams;
+        }
+
+        private CompiledRoute(Route baseRoute, String ratelimitRoute, String compiledRoute)
+        {
+            this(baseRoute, ratelimitRoute, compiledRoute, false);
+        }
+
+        public CompiledRoute withQueryParams(String name, String value)
+        {
+            return new CompiledRoute(baseRoute, ratelimitRoute, compiledRoute + (hasQueryParams ? '&' : '?') + name + '=' + value, true);
+        }
+
+        public CompiledRoute withQueryParams(String... params)
+        {
```

Comment:
What happens if you call it like: `withQueryParams()` ? Looks like it would set `hasQueryParams` to true, and use `&` from then on, which would be wrong.

Additionally, if we have a `String...` version, do we really need a `(String, String)` version? Especially since we're making sure there is a multiple of two params?